sql/descriptor_cache_test.go

```diff
@@ -0,0 +1,241 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Marc Berhault (marc@cockroachlabs.com)
+
+package sql_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/cockroachdb/cockroach/client"
+	"github.com/cockroachdb/cockroach/config"
+	"github.com/cockroachdb/cockroach/gossip"
+	"github.com/cockroachdb/cockroach/keys"
+	"github.com/cockroachdb/cockroach/server"
+	"github.com/cockroachdb/cockroach/sql"
+	"github.com/cockroachdb/cockroach/util"
+	"github.com/cockroachdb/cockroach/util/leaktest"
+	gogoproto "github.com/gogo/protobuf/proto"
+)
+
+var configDesc = sql.DatabaseDescriptor{Name: "sentinel", ID: 1, Privileges: &sql.PrivilegeDescriptor{}}
+var configDescKey = sql.MakeDescMetadataKey(keys.MaxReservedDescID)
+
+// waitForConfig forces a system config update by writing a bogus descriptor with an
+// incremented value inside. It then repeatedly fetches the gossip config until the
+// just-written descriptor is found.
+func waitForConfig(t *testing.T, s *server.TestServer) (*config.SystemConfig, error) {
+	configDesc.ID += 1
```

Comment:
I can definitely build the descriptor itself every time. However, the ID is intentionally global to avoid the tests having to know about it.